roboflow/adapters/rfapi.py

```diff
@@ -158,22 +158,37 @@ def _save_annotation_url(api_key, project_url, name, image_id, job_name, is_pred
     return url
 
 
-def _hosted_upload_url(api_key, project_url, image_path, split):
+def _upload_url(api_key, project_url, **kwargs):
     url = f"{API_URL}/dataset/{project_url}/upload?api_key={api_key}"
-    url += f"&name={os.path.basename(image_path)}&split={split}"
-    url += f"&image={urllib.parse.quote_plus(image_path)}"
+
+    for key, value in kwargs.items():
+        if isinstance(value, list):
+            url += "".join([f"&{str(key)}={str(v)}" for v in value])
+        else:
+            url += f"&{str(key)}={str(value)}"
+
     return url
 
 
+def _hosted_upload_url(api_key, project_url, image_path, split, batch_name, tag_names):
+    return _upload_url(
+        api_key,
+        project_url,
+        name=os.path.basename(image_path),
+        split=split,
+        image=urllib.parse.quote_plus(image_path),
+        batch_name=batch_name,
+        tag=tag_names,
+    )
+
+
 def _local_upload_url(api_key, project_url, batch_name, tag_names, sequence_number, sequence_size, kwargs):
-    url = f"{API_URL}/dataset/{project_url}/upload?api_key={api_key}&batch={batch_name}"
+    query_params = dict(batch_name=batch_name, tag=tag_names, **kwargs)
```

Comment:
The URL used a `batch` param instead of `batch_name`. Is the param wrong or you confused names here? I think `batch` is correct, but didn't investigate the API.

```suggestion
    query_params = dict(batch=batch_name, tag=tag_names, **kwargs)
```

Reply:
Great catch, fixed at 21a26e21673285dee6884e6df3e8d5251ca0b79f